### PR TITLE
[enchement](mc)Optimize reading of maxcompute partition tables.

### PR DESCRIPTION
**Pull request by hubgeter**
### What problem does this PR solve?
Problem Summary:
Optimize reading of maxcompute partition tables:
1. Introduce batch mode to generate splits for Maxcompute partition tables to optimize scenarios with a large number of partitions. Control it through the variable `num_partitions_in_batch_mode`.
2. Introduce catalog parameter `mc.split_cross_partition`. The parameter is true, which is more friendly to reading partition tables, and false, which is more friendly to debug.
3. Add `-Darrow.enable_null_check_for_get=false` to be jvm to improve the efficiency of mc arrow data conversion.

### Release note

None

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [ ] No.
    - [ ] Yes. 

- Does this need documentation?
    - [ ] No.
    - [x] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by hubgeter** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/datasource/SplitAssignment.java [hunk 1]
```diff
@@ -156,7 +156,7 @@ public BlockingQueue<Collection<TScanRangeLocations>> getAssignedSplits(Backend
         return splits;
     }
 
-    public void setException(UserException e) {
+    public synchronized void setException(UserException e) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/SplitAssignment.java:159]: Why using `synchronized `? add comment in code

File: fe/fe-core/src/main/java/org/apache/doris/datasource/maxcompute/source/MaxComputeScanNode.java [hunk 2]
```diff
@@ -181,32 +174,118 @@ boolean createTableBatchReadSession() throws UserException {
...
+                    break;
+                }
+                List<PartitionSpec> requiredBatchPartitionSpecs = requiredPartitionSpecs.subList(beginIndex, endIndex);
+                int curBatchSize = endIndex - beginIndex;
+
+                try {
+                    CompletableFuture.runAsync(() -> {
+                        try {
+                            TableBatchReadSession tableBatchReadSession =
+                                    createTableBatchReadSession(requiredBatchPartitionSpecs);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/maxcompute/source/MaxComputeScanNode.java:264]: Looks like in this new implementation, you create read session for each partition?
Is `create read session` a heavy operation?

Comment (hubgeter) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/maxcompute/source/MaxComputeScanNode.java:264]: create read session for a batch of partition . `createTableBatchReadSession`  have network io , and increasing the number of partitions will cause network io to be very slow. I tested 1500 partitions and it took about 13 seconds.

File: fe/fe-core/src/main/java/org/apache/doris/datasource/SplitAssignment.java (hunk 1 above)
Comment (hubgeter) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/SplitAssignment.java:159]: It seems that there is no need to add synchronized.

**Comment by hubgeter** [comment]
run buildall

**Review by morningman — APPROVED** [review]

**Review by zy-kkk — APPROVED** [review]

**Pull request merged**